project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/sample/openbis/OpenbisConnector.java

```diff
@@ -266,27 +258,104 @@ private void createOpenbisSamples(List<SampleCreation> samplesToRegister) {
     handleOperations(operation);
   }
 
+  private void updateOpenbisSamples(List<SampleUpdate> samplesToUpdate) {
+    IOperation operation = new UpdateSamplesOperation(samplesToUpdate);
+    handleOperations(operation);
+  }
+
   /**
-   * Deletes a sample with the provided code from persistence.
+   * Deletes a collection of samples with the provided codes from persistence. Checks if any of the
+   * samples has attached data and fails the deletion of the sample batch, if so.
    *
-   * @param sampleCode the {@link SampleCode} of the sample to delete
+   * @param projectCode the {@link ProjectCode} of the project these samples belong to
+   * @param samples The {@link Sample}s to be deleted in the data repo
+
    * @since 1.0.0
    */
   @Override
-  public void delete(SampleCode sampleCode) {
-    deleteOpenbisSample(DEFAULT_SPACE_CODE, sampleCode.code());
+  public void deleteAll(ProjectCode projectCode,
+      Collection<life.qbic.projectmanagement.domain.model.sample.Sample> samples) {
+
+    Set<String> sampleCodesToDelete = samples.stream().map(sample -> sample.sampleCode().code())
+        .collect(Collectors.toSet());
+
+    //Fetch samples with potential data - sample search is not working, sorry you have to see this
+    ExperimentFetchOptions fetchOptions = new ExperimentFetchOptions();
+    fetchOptions.withSamplesUsing(fetchSamplesCompletely());
+    for (Experiment experiment : searchExperimentsByProjectCode(projectCode.value(), fetchOptions)) {
```

Comment:
Please Refactor!!!
You can use inverted if statements and earlier exit to prevent this chaining of if statements.